### PR TITLE
Bookend tests with latest & oldest Python dependencies

### PR DESCRIPTION
**Pull request by jared321**
Need a PR to get actions running.

edit from (@mosesyhc): 
This PR continues the effort of #134 as a clean PR after the merge/revert problems.
The file changes in #134 should be considered as well as the ones in here.
- This PR mainly contains bookend tests and `test_new_cal`, `test_new_emu` test suite modules. 
- #134 mainly contains `test_emu_cal` test suite module.

### CODE REVIEW

**Comment by mosesyhc** [comment]
@jared321: 
To me, this PR is ready for v0.3.

Instead of asking for a full review for this PR, I suggest merging and request for a full review on #121, which contains the combined changes of the collection of branches relevant to updating the test suite.

**Comment by jared321** [comment]
This PR has 12 files changed.  For me that's already a somewhat large PR, which can make it harder for a reviewer to find all issues, especially subtle issues or single changes that span multiple files.  The review of #121 will be even harder since it is massive.  In my mind, that review will be most effective at trying to get the global picture of all that was done, checking that all necessary tasks were performed, checking for issues at that level, rechecking at a low-level as an extra level of security, and convincing ourselves that the release is ready.

If you don't mind, I'll review this as best as I can here as the "official" review and then re-review with #121 as more of a sanity check.

Food for thought, what would these layers of checks look like if your workflow has release branches that don't merge into a branch like `main`?

**Comment by mosesyhc** [comment]
I do not mind at all. Please review this PR as best as we can.

I intended to reduce the number of reviews one has to do, but I agree with your analysis about the size of this PR and #121 being even larger.  It may be then helpful to outline also the expectation as we move forward to reviewing #121.

My impression is that after a release is done, PRs to those release branches should be minimal / patches, instead of any big changes of functionality like in this PR. If such changes are indeed needed, then there should be a new release, instead of working on that release branch.

________________________________

**Review by jared321 — COMMENTED** [review]
Review done with two commits.  Just a few comments/questions.

File: surmise/calibrationmethods/mlbayeswoodbury.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def thetalpdf(fitinfo, theta, args=None):
     return (logpost-fitinfo['lpdfapproxnorm'])
 
 
-def loglik(fitinfo, emu, theta, y, x):
+def loglik(fitinfo, emu, theta, y, x, args=None):
```
[the remark this answers is not in the document]
Comment (mosesyhc) [reply surmise/calibrationmethods/mlbayeswoodbury.py:297]: This is for reconciling arguments passed down from a higher-level code.

**Pull request merged**